superset/migrations/versions/2024-02-14_14-43_17fcea065655_change_text_to_mediumtext.py

```diff
@@ -0,0 +1,84 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+"""change_text_to_mediumtext
+
+Revision ID: 17fcea065655
+Revises: 87d38ad83218
+Create Date: 2024-02-14 14:43:39.898093
+
+"""
+
+# revision identifiers, used by Alembic.
+revision = "17fcea065655"
+down_revision = "87d38ad83218"
+
+import sqlalchemy as sa
+from alembic import op
+
+from superset.utils.core import MediumText
+
+TABLE_COLUMNS = [
+    "annotation.json_metadata",
+    "css_templates.css",
+    "dashboards.css",
+    "keyvalue.value",
+    "query.extra_json",
+    "query.executed_sql",
+    "query.select_sql",
+    "report_execution_log.value_row_json",
+    "report_recipient.recipient_config_json",
+    "report_schedule.sql",
+    "report_schedule.last_value_row_json",
+    "report_schedule.validator_config_json",
+    "report_schedule.extra_json",
+    "row_level_security_filters.clause",
+    "saved_query.sql",
+    "saved_query.extra_json",
+    "sl_columns.extra_json",
+    "sl_datasets.extra_json",
+    "sl_tables.extra_json",
+    "slices.params",
+    "slices.query_context",
+    "ssh_tunnels.extra_json",
+    "tab_state.extra_json",
+    "tab_state.sql",
+    "table_schema.extra_json",
+]
+
+NOT_NULL_COLUMNS = ["keyvalue.value", "row_level_security_filters.clause"]
+
+
+def upgrade():
+    for column in TABLE_COLUMNS:
+        with op.batch_alter_table(column.split(".")[0]) as batch_op:
```

Comment:
@michael-s-molina, I encountered a database deadlock error when running the database migration script a few days ago. We are using PostgreSQL as Superset metadata. This change only affects MySQL, so could you please add a condition to detect if MySQL is used as metadata? If not, pass the upgrade and downgrade functions.

Reply:
@zhaoyongjie are these migrations a no-op if PostgreSQL doesn't differentiate between `TEXT` and `MEDIUMTEXT`?

Reply:
@john-bodley PG only has TEXT type for string-like column, I'm not sure what should be done under the SQLAlchemy core.

Reply:
@john-bodley According to ChatGPT:

> No, the alter_column function in SQLAlchemy is not a no-op if the existing type is the same as the new type. When you use the alter_column function in SQLAlchemy to alter a column, it will generate an ALTER TABLE statement that modifies the column definition in the database schema. Even if the existing type is the same as the new type, SQLAlchemy will still execute the ALTER TABLE statement to ensure that the column definition matches the intended type.
This behavior ensures consistency and helps to handle any other modifications or changes associated with the column, such as constraints or defaults.